modules/javafx.base/src/main/java/javafx/beans/value/ObservableValue.java

```diff
@@ -137,4 +139,55 @@
      * @return The current value
      */
     T getValue();
+
+    /**
+     * Returns an {@link ObservableValue} which provides a mapping of the value
+     * held by this {@code ObservableValue}, and provides {@code null} when this
+     * {@code ObservableValue} holds {@code null}.
+     *
+     * @param <U> the type of values held by the resulting {@code ObservableValue}
+     * @param mapper a {@link Function} which converts a given value to a new value, cannot be null
+     * @return an {@link ObservableValue} which provides a mapping of the value
+     *     held by this {@code ObservableValue}, and provides {@code null} when
+     *     this {@code ObservableValue} holds {@code null}, never null
+     */
+    default <U> ObservableValue<U> map(Function<? super T, ? extends U> mapper) {
+        return new MappedBinding<>(this, mapper);
+    }
+
+    /**
+     * Returns an {@link ObservableValue} which provides a mapping of the value
+     * held by this {@code ObservableValue}, or {@code constant} when this
+     * {@code ObservableValue} holds {@code null}.
+     *
+     * @param constant an alternative value to use when this {@code ObservableValue}
+     *     holds {@code null}, can be null
+     * @return an {@link ObservableValue} which provides a mapping of the value
+     *     held by this {@code ObservableValue}, or {@code constant} when this
+     *     {@code ObservableValue} holds {@code null}, never null
+     */
+    default ObservableValue<T> orElse(T constant) {
+        return new OrElseBinding<>(this, constant);
+    }
+
+    /**
+     * Returns an {@link ObservableValue} which provides the value in the {@code
+     * ObservableValue} given by applying {@code mapper} on the value held by this
+     * {@code ObservableValue}, and is {@code null} when this
+     * {@code ObservableValue} holds {@code null}.<p>
+     *
+     * Returning {@code null} from {@code mapper} will result in an
+     * {@code ObservableValue} which holds {@code null}.
+     *
+     * @param <U> the type of values held by the resulting {@code ObservableValue}
+     * @param mapper a {@link Function} which converts a given value to an
+     *     {@code ObservableValue}, cannot be null
```

Comment:
`a {@code Function} that converts a given value to an {@code ObservableValue}; cannot be {@code null}`